tests/conftest.py

```diff
@@ -28,27 +22,21 @@
 #                                                                              #
 ################################################################################
 
-import sys
-import unittest
+from . import Framework
 
-from . import AllTests, Framework
 
+def pytest_addoption(parser):
+    parser.addoption("--record", action="store_true", help="record mode")
+    parser.addoption(
+        "--auth_with_token", action="store_true", help="auth using a token"
+    )
+    parser.addoption("--auth_with_jwt", action="store_true", help="auth using JWT")
 
-def main(argv):
-    if "--record" in argv:
-        Framework.activateRecordMode()
-        argv = [arg for arg in argv if arg != "--record"]
 
-    if "--auth_with_token" in argv:
+def pytest_configure(config):
+    if config.getoption("record"):
+        Framework.activateRecordMode()
+    if config.getoption("auth_with_token"):
         Framework.activateTokenAuthMode()
-        argv = [arg for arg in argv if arg != "--auth_with_token"]
-
-    if "--auth_with_jwt" in argv:
+    if config.getoption("auth_with_jwt"):
```

Comment:
Does the record and replay testing framework still work? 

Maybe we should also update the docs: https://github.com/PyGithub/PyGithub/blob/08b1c474b32d3f32bc64a59a7e77d87c43527e1f/CONTRIBUTING.md#L73

Reply:
Embarrassing truth time: I don't use it. However, I did check that the right methods in Framework are called if you call pytest with --record and such.

Reply:
Excellent point about updating the docs, I'm doing so.

Reply:
I see, anyway we should update all reference of `python -m test` to `pytest`

Reply:
Am I being judged for using pdb and list comprehensions in a text editor to generate my replay date? :laughing: